### ISSUE TITLE
Checkbox column rendered incorrectly after removing filter

### ISSUE THREAD

**Issue by bmblb** [bryntum/support#8541]
Reported by a client.

https://github.com/bryntum/support/assets/45400762/275d23a1-9624-460a-9eef-575a437966d8

Steps to reproduce on grid filtering demo:
1. replace app code with a snippet below
2. click `toggle selected` twice

Notice all checkboxes in the first columns are disabled and cannot be checked.

```
import { Grid } from "../../build/grid.module.js?474079";

import shared from "../_shared/shared.module.js?474079";

const handleSelectionChange = ({ checked, record }) => {
    if (!checked) {
        grid.store.records.forEach((r) => {
            if (!record || record.id === r.id) {
                r.useReport = r.useFilters = r.useComments = false;
            }
        });
    }
};

let showSelected = false;

const grid = new Grid({
    appendTo: "container",

    tbar: [
        {
            ref: "toggleSelected",
            text: "Toggle Selected",
            onAction: () => {
                showSelected = !showSelected;

                grid.store.filter((row) => {
                    return showSelected ? row.useReport : true
                });
            },
        },
    ],

    columns: [
        {
            showCheckAll: true,
            field: "useReport",
            type: "check",
            width: 60,
            renderer({ record, row, cellElement }) {
                // Highlight selected rows
                if (record.useReport) {
                    row.addCls("b-selected");
                }

                // Appear clean
                cellElement.classList.remove("b-cell-dirty");
            },
            listeners: {
                toggle: handleSelectionChange,
                toggleAll: handleSelectionChange
            },
        },
        {
            text: "Report Name",
            field: "name",
            flex: 2,
            editor: false,
        },
        {
            text: "Apply Filters",
            field: "useFilters",
            type: "check",
            flex: 1,
            renderer({ record, widgets, cellElement }) {
                // Disable on reports with no filters
                widgets[0].disabled = !record.hasFilters || !record.useReport;

                // Appear clean
                cellElement.classList.remove("b-cell-dirty");
            },
        },
        {
            text: "Include Comments",
            field: "useComments",
            type: "check",
            flex: 1,
            renderer({ record, widgets, cellElement }) {
                // Disable on reports with no comments
                widgets[0].disabled = !record.hasComments || !record.useReport;

                // Appear clean
                cellElement.classList.remove("b-cell-dirty");
            },
        },
    ],

    data: [{
        name: 'Test Report 1',
        useReport: false,
        useFilters: false,
        useComments: false,
    }, {
        name: 'Test Report 2',
        useReport: false,
        useFilters: false,
        useComments: false,
    }, {
        name: 'Test Report 3',
        useReport: false,
        useFilters: false,
        useComments: false,
    }],
});
```

**Comment by amurashincertinia** [comment]
If it helps, the example snippet works as expected with version `5.6.4` but breaks with `5.6.5` and `5.6.6`.

**Comment by bmblb** [comment]
This one is fixed along with #8484
